src/main/java/org/minions/devfund/angela/Squarelotron.java

```diff
@@ -0,0 +1,179 @@
+            for (int j = 0; j < getSize(); j++) {
+
+                numberArray[i * getSize() + j] = getSquarematrix()[i][j];
+            }
+        }
+        return numberArray;
+    }
+
+    /**
+     * Performs the Upside-Down Flip of the squarematrix.
+     *
+     * @param ring int ring number to perform the Upside-Down Flip.
+     * @return {@link Squarelotron}.
+     */
+    Squarelotron upsideDownFlip(final int ring) {
+        int[] numbers = changeToUnidimensionalArray();
+        int[] newNumbers = new int[getSize() * getSize()];
+        for (int i = 0; i < getSize(); i++) {
+            for (int j = 0; j < getSize(); j++) {
+                if (i == ring - 1 || i == getSize() - ring || j == ring - 1 || j == getSize() - ring) {
+                    newNumbers[i * getSize() + j] = numbers[(getSize() - i - 1) * getSize() + j];
+                } else {
+                    newNumbers[i * getSize() + j] = numbers[i * getSize() + j];
+                }
+            }
+        }
+        return new Squarelotron(newNumbers);
+    }
+
+
+    /**
+     * Performs the Main Diagonal Flip of the squarematrix.
+     *
+     * @param ring int ring number to perform the Main Diagonal Flip.
+     * @return {@link Squarelotron}.
+     */
+    public Squarelotron mainDiagonalFlip(final int ring) {
+        int[] numbers = changeToUnidimensionalArray();
+        int[] newNumbers = new int[getSize() * getSize()];
+        for (int i = 0; i < getSize(); i++) {
+            for (int j = 0; j < getSize(); j++) {
+                if (i == ring - 1 || i == getSize() - ring || j == ring - 1 || j == getSize() - ring) {
+                    newNumbers[i * getSize() + j] = numbers[j * getSize() + i];
+                } else {
+                    newNumbers[i * getSize() + j] = numbers[i * getSize() + j];
+                }
+            }
+        }
+        return new Squarelotron(newNumbers);
+    }
+
+    /**
+     * Rotates the squarematrix to right.
+     *
+     * @param numberOfTurns indicates the number of times the entire
+     *                      squarematrix should be rotated 90° clockwise .
+     */
+    public void rotateRight(final int numberOfTurns) {
+        final int squareSides = 4;
+        final int oneRotateRight = 1;
+        final int oneRotateLeft = -1;
+        final int twoRotateRight = 2;
+        final int twoRotateLeft = -2;
+        final int threeRotateRight = 3;
+        final int threeRotateLeft = -3;
+        if (numberOfTurns % squareSides == oneRotateRight || numberOfTurns % squareSides == threeRotateLeft) {
+            rotate();
+
+        } else if (numberOfTurns % squareSides == twoRotateRight || numberOfTurns % squareSides == twoRotateLeft) {
+            rotate();
+            rotate();
+
+        } else if (numberOfTurns % squareSides == threeRotateRight || numberOfTurns % squareSides == oneRotateLeft) {
+            rotate();
+            rotate();
+            rotate();
```

Comment:
Instead of doing this non maintainable method.
You can call the first method and then the second one as your teammate suggested.